Framework/Geometry/src/Instrument/Component.cpp

```diff
@@ -393,6 +387,152 @@ Quat Component::getRotation() const {
       return m_parent->getRotation() * m_rot;
   }
 }
+*/
+
+/// Helper for legacy access mode. Returns the index of the component.
+size_t Component::index() const {
+  return m_map->componentIndex(this->getComponentID());
+}
+
+bool Component::hasComponentInfo() const {
+  const IComponent *root = m_base;
+  while (auto parent = root->getBareParent())
+    root = parent;
+  auto instrument = dynamic_cast<const Instrument *>(root);
+  return m_map->hasComponentInfo(instrument);
+}
+
+/// Return the relative position to the parent Component
+Kernel::V3D Component::getRelativePos() const {
+  if (m_map) {
+
+    if (hasComponentInfo() && !m_map->componentInfo().isDetector(index())) {
```

Comment:
I initially thought that all old parameter map code in `Component::getPos` and `DetectorInfo::getPos` etc could be made redundant if we were to lean on `ComponentInfo` methods such as `ComponentInfo::getRelativeRotation`, which means establishing the `ComponentInfo` first. In reality, it doesn't matter. As we know that the `InfoComponentVisitor` can and will be used to parse and flush the whole instrument tree in one go (so we effectively get the `DetectorInfo` and `ComponentInfo` in one hit). I suggest that it doesn't matter at this stage as long as we can preserve working functionality in the intermediate builds.